adminapi/dataset.py

```diff
@@ -230,6 +231,32 @@ def get_free_ip_addrs(self):
                     if host not in used_hosts:
                         yield host
 
+    def get_free_ip_addr(self, lock=True):
+        """Get one free IP address from network
+
+        This will give you exactly one free IP address for the queried network
+        and lock it to avoid somebody else using it. If you do not care you
+        can set lock to false but may need to handle the CommitError with the
+        duplicate IP on your own.
+
+        :param lock: Lock free IP address for 60 seconds
+
+        :return:
+        """
+
+        if not lock:
+            return next(self.get_network_ip_addrs())
```

Comment:
However, this can actually produce a `StopIteration` exception if the generator/iterator does not have any `next`. You might want to catch that here then.

Reply:
Indeed a more speaking exception might be helpful - will adjust.